kafka-client/src/main/java/dev/responsive/kafka/internal/db/rs3/RS3TableFactory.java

```diff
@@ -14,38 +14,79 @@
 
 import dev.responsive.kafka.api.config.ResponsiveConfig;
 import dev.responsive.kafka.internal.db.RemoteKVTable;
+import dev.responsive.kafka.internal.db.rs3.client.CreateStoreOptions;
+import dev.responsive.kafka.internal.db.rs3.client.CreateStoreOptions.ClockType;
 import dev.responsive.kafka.internal.db.rs3.client.WalEntry;
 import dev.responsive.kafka.internal.db.rs3.client.grpc.GrpcRS3Client;
 import dev.responsive.kafka.internal.metrics.ResponsiveMetrics;
+import dev.responsive.kafka.internal.stores.TtlResolver;
+import java.util.HashSet;
 import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
 import java.util.UUID;
+import java.util.function.Supplier;
 import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.streams.errors.StreamsException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class RS3TableFactory {
+  private static final Logger LOG = LoggerFactory.getLogger(RS3TableFactory.class);
   private final GrpcRS3Client.Connector connector;
 
+  // kafka store names to track which stores we've created in RS3
+  private final Set<String> createdStores = new HashSet<>();
+
   public RS3TableFactory(GrpcRS3Client.Connector connector) {
     this.connector = connector;
   }
 
   public RemoteKVTable<WalEntry> kvTable(
-      final String name,
+      final String storeName,
       final ResponsiveConfig config,
+      final Optional<TtlResolver<?, ?>> ttlResolver,
       final ResponsiveMetrics responsiveMetrics,
-      final ResponsiveMetrics.MetricScopeBuilder scopeBuilder
+      final ResponsiveMetrics.MetricScopeBuilder scopeBuilder,
+      final Supplier<Integer> computeNumKafkaPartitions
   ) {
-    Map<String, String> storeIdMapping = config.getMap(
+    final Map<String, String> storeIdMapping = config.getMap(
         ResponsiveConfig.RS3_LOGICAL_STORE_MAPPING_CONFIG);
-    final String storeIdHex = storeIdMapping.get(name);
+    final String storeIdHex = storeIdMapping.get(storeName);
     if (storeIdHex == null) {
-      throw new ConfigException("Failed to find store ID mapping for table " + name);
+      throw new ConfigException("Failed to find store ID mapping for table " + storeName);
     }
 
     final UUID storeId = UUID.fromString(storeIdHex);
-    final PssPartitioner pssPartitioner = new PssDirectPartitioner();
+
     final var rs3Client = connector.connect();
+
+    if (!createdStores.contains(storeName)) {
+      final int kafkaPartitions = computeNumKafkaPartitions.get();
+
+      final Optional<Long> defaultTtl = ttlResolver.isPresent() && ttlResolver.get().defaultTtl().isFinite()
+          ? Optional.of(ttlResolver.get().defaultTtl().duration().toMillis())
+          : Optional.empty();
+
+      final var options = new CreateStoreOptions(
+          ttlResolver.isPresent() ? Optional.of(ClockType.WALL_CLOCK) : Optional.empty(),
+          defaultTtl,
+          Optional.empty()
```

Comment:
this is the filterBitsPerKey, not sure if we'd want to make this a public/Responsive config? Have we thought at all about how to handle slateDB configuration yet?

Reply:
I think we can just take the configs on a case-by-case basis and add a config property for the ones where we think it's appropriate. I wouldn't really expect users to configure this particular setting. I'd just let this be the default and then allow it to be overrided for specific stores (the one I have in mind is the fact store where we want to set it to something high (I'm thinking 20 bits) to optimize for dedupers)

Reply:
ah I didn't realize we were going to implement a fact store for rs3. have you put any thought into what exactly it would look like yet? ie any difference besides the filter bits value?

Reply:
Nah that was the main difference in my mind for the near term. The only other thing I can think of that we may want to tweak is the compaction algorithm. We can probably take advantage of the fact that it's insert-only and time-ordered. But that's for later.

Reply:
ah ok. was just wondering about this since discussing how to support window stores in CreateStore. Sounds like we should add a "storetype" field that combines the "schema" enums for all store types, so the options would be kv, fact, window, window-with-duplicates, and session